### DISCUSSION TITLE
Rewrites not working when deployed to ECS

### DISCUSSION THREAD

**Discussion by gyopiazza** [vercel/next.js#61319] [Help]
### Summary

Hello! I have a rewrite rule in `next.config.js` which works locally, but it doesn't when deployed on AWS ECS. The app uses SSR.

### Additional information

This is the configuration for the rewrite which works locally:

```js
rewrites: () => [
    {
      source: '/.well-known/apple-developer-merchantid-domain-association',
      destination: `/apple-developer-merchantid-domain-association/${process.env.ENVIRONMENT}`,
    },
  ],
```

Note: the public files are correctly deployed and are accessible, the `ENVIRONMENT` is also correctly set.

Has anybody else experienced this?
Thank you in advance

**Comment by icyJoseph** [comment]
Hi,

Are you including the next.config.js file in the deployed runtime? I don't remember if it is one of those things that also has to be included in the deployed version.

**Reply by gyopiazza** [reply]
Thanks for the quick reply @icyJoseph !

Other configuration options (eg `redirects`) work correctly once deployed, the problem seems to be just with `rewrites`. Or maybe I'm missing something?

**Reply by icyJoseph** [reply]
MM I wonder if you have to use named keys, like, beforeFiles, etc:

```markdown
When the rewrites function returns an array, rewrites are applied after checking the filesystem (pages and /public files) and before dynamic routes
```

- https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites

**Reply by gyopiazza** [reply]
It's strange that it works locally though, but I'll make a some tests and let you know. Thanks!

**Reply by gyopiazza** [reply]
I've made several tests but nothing seems to work for rewrites. Any other ideas?

**Reply by icyJoseph** [reply]
Not really, this is quite strange. Perhaps there's some headers that are getting dropped?

**Comment by victormattosvm** [comment]
I guess that's because the next.config.js is not being copied to the docker image. Check your dockerfile.
